clang/include/clang/Serialization/TypeBitCodes.def

```diff
@@ -63,4 +63,7 @@ TYPE_BIT_CODE(DependentExtInt, DEPENDENT_EXT_INT, 51)
 TYPE_BIT_CODE(ConstantMatrix, CONSTANT_MATRIX, 52)
 TYPE_BIT_CODE(DependentSizedMatrix, DEPENDENT_SIZE_MATRIX, 53)
 
+// CHERI extension; leave room for new upstream IDs
+TYPE_BIT_CODE(DependentPointer, DEPENDENT_POINTER, 80) // 'P'
```

Comment:
I wonder if we could get this upstream by supporting __ptr32/__ptr64. However, MSVC doesn't seem to handle it on templates either, so upstream might not be too keen on the added complexity.

Reply:
That could be an interesting way to reduce our diff, though I'm not sure it's worth the effort of unifying the existing ptrXX support in Clang with our PIK stuff. If someone wants to then I'm all for it, but it's not something I have time for.

Reply:
Also if MSVC doesn't support it then that means there isn't a defined mangling for it.